### PR TITLE
refactor: 抽取属性设置面板布局组件

### PR DESCRIPTION
**Pull request by gene9831**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Added support for new settings panels and layout settings in the design core module.
  - Introduced dynamic component retrieval based on metadata in the design core's main application.

- **Bug Fixes**
  - Corrected identifier names across various modules to ensure consistency and functionality.

- **Improvements**
  - Enhanced the `Main.vue` component in the settings panel to support dynamic tab rendering and improved state management.
  - Updated Vite configuration files to include new plugin aliases and build settings for better project organization and build processes.

### CODE REVIEW

File: packages/design-core/src/App.vue [hunk 1]
```diff
@@ -10,15 +10,22 @@
           </div>
         </div>
         <div class="tiny-engine-right-wrap">
-          <design-settings v-show="layoutState.settings.showDesignSettings" ref="right"></design-settings>
+          <component
```
Comment (chilingling) [inline packages/design-core/src/App.vue]: 这里跟全局的 layout 替换感觉会有冲突？可以放到 `DesignSettings.vue` 组件里面？

File: packages/layouts/settings-panel/meta.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
+ * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
+ * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
+ *
+ */
+
+export default {
+  id: 'engine.layouts.settings-panel',
+  title: '',
```
Comment (chilingling) [inline packages/layouts/settings-panel/meta.js]: 这个 title 没有用就先不加吧，用到了再加

File: packages/layouts/settings-panel/vite.config.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [generateComment(), vue(), vueJsx()],
+  publicDir: false,
+  resolve: {},
+  build: {
+    lib: {
+      entry: path.resolve(__dirname, './index.js'),
+      name: 'layout-settings-panel',
```
Comment (chilingling) [inline packages/layouts/settings-panel/vite.config.js]: `format` 只有一个 `es` 的话，好像不需要加 `name`。只有 umd 和 cjs 的格式需要

File: packages/layouts/settings-panel/meta.js (hunk 2 above)
Comment (gene9831) [reply packages/layouts/settings-panel/meta.js]: 已移除

File: packages/layouts/settings-panel/vite.config.js (hunk 3 above)
Comment (gene9831) [reply packages/layouts/settings-panel/vite.config.js]: 已修改

File: packages/design-core/src/App.vue (hunk 1 above)
Comment (gene9831) [reply packages/design-core/src/App.vue]: done

File: packages/settings/panel/vite.config.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * Use of this source code is governed by an MIT-style license.
+ *
+ * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
+ * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
+ * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
+ *
+ */
+
+import { defineConfig } from 'vite'
+import path from 'path'
```
[the remark this answers is not in the document]
Comment (gene9831) [reply packages/settings/panel/vite.config.js]: updated

File: packages/layouts/settings-panel/vite.config.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ * Use of this source code is governed by an MIT-style license.
+ *
+ * THE OPEN SOURCE SOFTWARE IN THIS PRODUCT IS DISTRIBUTED IN THE HOPE THAT IT WILL BE USEFUL,
+ * BUT WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS FOR
+ * A PARTICULAR PURPOSE. SEE THE APPLICABLE LICENSES FOR MORE DETAILS.
+ *
+ */
+
+import { defineConfig } from 'vite'
+import path from 'path'
```
[the remark this answers is not in the document]
Comment (gene9831) [reply packages/layouts/settings-panel/vite.config.js]: updated

File: packages/design-core/src/DesignSettings.vue [hunk 6]
```diff
@@ -1,107 +1,25 @@
...
-    const activating = computed(() => layoutState.settings.activating)
-    const showMask = ref(true)
 
-    return {
-      showMask,
-      settings,
-      activating,
-      layoutState
-    }
+    return { SettingsPanel, layoutState }
```
[the remark this answers is not in the document]
Comment (gene9831) [reply packages/design-core/src/DesignSettings.vue:22]: template中使用了判断保证不会报错，不过缺少fallback页面。之后考虑统一添加fallback页面

File: packages/settings/panel/src/Main.vue [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+  },
+
+  setup() {
+    const settings = getMergeMeta('engine.settings')?.metas || []
+
+    return {
+      settings
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (gene9831) [reply packages/settings/panel/src/Main.vue]: settings可以为空数组，不需要输出错误

File: packages/settings/panel/src/Main.vue [hunk 8]
```diff
@@ -0,0 +1,108 @@
+<template>
+  <div id="tiny-right-panel">
```
Comment (chilingling) [inline packages/settings/panel/src/Main.vue:2]: 这个 id 跟样式、全局的布局可能还有一些耦合或者关联，到时还需要注意一下

**Review by chilingling — APPROVED** [review]

**Pull request merged**